src/audio/volume/volume.c

```diff
@@ -311,6 +329,11 @@ static inline void volume_ramp(struct processing_module *mod)
 		cd->volume[i] = new_vol;
 	}
 
+	if (cd->is_same_volume) {
+		for (i = 1; i < cd->channels; i++)
```

Comment:
I'd just do `for (i = cd->ramp_channel_counter; i < cd->channels; i++)` and remove the `if` in line 332